### PR TITLE
(FM-2626) Add DSC Beaker Helper Library

### PR DESCRIPTION
**Pull request by cowofevil**
Add a helper library for testing DSC. Also, I added one test to verify
that the helper library actually works and a test runner script.

### CODE REVIEW

File: tests/integration/pre-suite/01_dsc_module_install.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+test_name 'FM-2626 - C68506 - Plug-in Sync Module from Master with Prerequisites Satisfied on Agent'
+
+confine(:to, :platform => 'windows')
```
Comment (zreichert) [inline tests/integration/pre-suite/01_dsc_module_install.rb]: Remove confine

File: tests/integration/pre-suite/00_pe_install.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+test_name 'CODEMGMT-20 - C48 - Install Puppet Enterprise'
```
Comment (zreichert) [inline tests/integration/pre-suite/00_pe_install.rb]: Wrong ticket number

File: tests/acceptance/tests/basic_functionality/puppet_apply_dsc_manifest.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  dsc_type            => 'File',
+  dsc_destinationpath => '#{test_file_path}',
+  dsc_contents        => '#{test_file_contents}'
+}
+MANIFEST
+
+#Teardown
+teardown do
+  step 'Remove Test Artifacts'
+  on(agents, "rm -rf #{test_dir_path}")
```
Comment (zreichert) [inline tests/acceptance/tests/basic_functionality/puppet_apply_dsc_manifest.rb]: Will not work, needs to be cygwin path.

File: tests/lib/dsc_utils.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Verify the expected state of a DSC resource on a given host.
+#
+# ==== Attributes
+#
+# * +host+ - The target Windows host for verification.
+# * +dsc_resource_type+ - The DSC resource type name to verify.
+# * +dsc_properties+ - A hash of DSC resource properties to verify.
```
Comment (zreichert) [inline tests/lib/dsc_utils.rb]: Order is incorrect

**Comment by cowofevil** [comment]
@zreichert Fixed, ready for review again.

**Pull request merged**